cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -75,10 +78,50 @@ func New(endpoints []string, ca, cert, key string) (*Client, error) {
 	return &client, nil
 }
 
+// IsEtcdListeningOnLocalHostOnly return true if the etcd manifest have etcd listening on localhost only.
+// Listening on local host only was the default in kubeadm <= v1.12, while starting from v1.13 etcd is listening
+// on localhost and API server advertise address (thus allowing add new member when doing join --control-plane).
+// N.B. this code is necessary only in v1.14; starting from v1.15 all the etcd manifest should have 2 endpoints
+func IsEtcdListeningOnLocalHostOnly() bool {
+	etcdManifestFile := constants.GetStaticPodFilepath(constants.Etcd, constants.GetStaticPodDirectory())
+	if _, err := os.Stat(etcdManifestFile); err == nil {
+		klog.V(1).Infoln("checking etcd manifest")
+		etcdPod, err := staticpod.ReadStaticPodFromDisk(etcdManifestFile)
+		if err == nil && len(etcdPod.Spec.Containers) > 0 {
+			etcdContainer := etcdPod.Spec.Containers[0]
+			for _, arg := range etcdContainer.Command {
+				if arg == "--listen-client-urls=https://127.0.0.1:2379" {
```

Comment:
Sorry this might be silly, but is there no way to have used a custom port and still have this issue?

Reply:
No, we don't provide supported ways of changing the port number here. It's hard wired to a constant (`EtcdListenClientPort`).
That said, we cannot stop users from editing the static pod specs, but we don't support such cases and users are on their own in that case.

Reply:
thank you.

Reply:
non blocking for merge.